core/config/options.go

```diff
@@ -71,6 +72,35 @@ func (opts *Options) List(ctx context.Context, key string) ([][]string, error) {
 	return tuples, nil
 }
 
+// ListFunc returns a closure that returns the set of tuples for the
+// provided key.
+//
+// The configuration option for key must be a set of tuples.
+// ListFunc will panic if the provided key is undefined in the schema or
+// is defined as a scalar. The returned function will perform a stale
+// read of the configuration value.
+func (opts *Options) ListFunc(key string) func() ([][]string, error) {
```

Comment:
It's too bad the closure has to return an error… I don't immediately see any way around it tho.